core/templates/pages/preferences-page/form-fields/subject-interests.component.ts

```diff
@@ -148,4 +148,26 @@ export class SubjectInterestsComponent implements ControlValueAccessor {
       interest.toLowerCase().includes(filterValue)
     );
   }
+
+  onBlur(): void {
+    const value = this.subjectInterestInput.nativeElement.value.trim();
+
+    if (value && this.validInput(value)) {
+      this.subjectInterests.push(value);
+      this.subjectInterestInput.nativeElement.value = '';
+    }
+    this.onChange(this.subjectInterests);
+  }
+
+  onInput(event: Event): void {
+    const inputElement = event.target as HTMLInputElement;
+    const inputValue = inputElement.value.trim();
+
+    if (!inputValue) {
+      this.formCtrl.markAsPristine();
+    } else if (this.validInput(inputValue)) {
+      this.formCtrl.markAsDirty();
+    }
```

Comment:
If the input is not valid then what should we do? (Also this case doesn't seem to be tested.)